src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationOptions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeGeneration
+{
+    internal readonly record struct CSharpCodeGenerationOptions(
+        CodeGenerationContext Context,
+        CSharpCodeGenerationPreferences Preferences)
```

Comment:
not exactly sure why we need Csharp version of this?  can this not be done with just a type in the shared lib?

Reply:
Allows use to move casts up to the overrides that implement language specific behavior and avoid casts in methods they call.